### PR TITLE
Remove use of `.device` in contrib.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Another sub-component for #4744 . This removes the use of `circuit.device` from all contrib modules.

BREAKING_CHANGE=removing the `device` property from the `SwapNetwork` class.
It also removes some type checking done by the code (all logic in the bulk still works), which makes it more forgiving.

With the amount of meaningful maintenance contrib has had (not much), I'm not a huge fan of going through a deprecation process for these mild behavior widenings with no API changes. Are others alright with this ?

### CODE REVIEW

**Review by 95-martin-orion — APPROVED** [review]
You may want to hunt down the author of `acquaintance` to confirm this.

Otherwise, no complaints from me on this - for a long time I thought we didn't even _test_ contrib, let alone make guarantees about stability, so I'm not too worried about deprecation-marking it.

**Comment by MichaelBroughton** [comment]
cc: @bryano

File: cirq-core/cirq/contrib/acquaintance/devices.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def validate_operation(self, operation: 'cirq.Operation') -> None:
 
 
 def is_acquaintance_strategy(circuit: 'cirq.Circuit') -> bool:
-    return isinstance(circuit._device, AcquaintanceDevice)
+    return True
```
Comment (dstrain115) [inline cirq-core/cirq/contrib/acquaintance/devices.py:45]: This seems weird and misleaeding that is_acquaintance_strategy is just always True.  Should we deprecate this function?

**Review by dstrain115 — APPROVED** [review]

File: cirq-core/cirq/contrib/acquaintance/devices.py (hunk 1 above)
Comment (MichaelBroughton) [reply cirq-core/cirq/contrib/acquaintance/devices.py:45]: It's not a part of the public API so nobody can call into this easily, but I'll go ahead and remove it.

**Pull request merged**